### ISSUE TITLE
Completion of error handling

### ISSUE THREAD

**Issue by elfring** [fdb/frequensea#4]
Would you like to add more error handling for return values from functions like the following?
- [fopen](http://pubs.opengroup.org/onlinepubs/9699919799/functions/fopen.html "open a stream") ⇒ [main](https://github.com/fdb/frequensea/blob/8e6af67da63d80e629875f65f979dcaff93ad302/c/render-text.c#L52)
- [printf](http://pubs.opengroup.org/onlinepubs/9699919799/functions/printf.html "print formatted output") ⇒ [receive_sample_block](https://github.com/fdb/frequensea/blob/8e6af67da63d80e629875f65f979dcaff93ad302/c/fft-batch.c#L54)

**Comment by fdb** [comment]
Absolutely! The error handling could be improved a lot. Any specific calls you've seen that are particularly important?

**Comment by elfring** [comment]
I suggest to avoid [ignorance of return values](https://cwe.mitre.org/data/definitions/252.html "Unchecked return values") a bit more.

Are you interested to apply [aspect-oriented software development](http://aspectc.org/ "AspectC++")?
How do you think about to encapsulate error detection and corresponding exception handling as [a reusable aspect](http://en.wikipedia.org/wiki/Aspect_%28computer_programming%29 "Combination of a pointcut with advice") in your software?
